src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java

```diff
@@ -383,20 +390,81 @@ public AdvisoryWrapper setDocumentTrackingStatus(DocumentTrackingStatus newState
 
-        final ObjectMapper jacksonMapper = new ObjectMapper();
-        ObjectNode versionNode = (ObjectNode) this.at(AdvisorySearchField.DOCUMENT);
-        ObjectNode trackingNode = (ObjectNode) versionNode.get("tracking");
-        if (trackingNode == null) {
-            trackingNode = jacksonMapper.createObjectNode();
-            versionNode.set("tracking", trackingNode);
-        }
+        ObjectNode trackingNode = getOrCreateTrackingNode();
         trackingNode.put("status", newState);
         return this;
     }
 
+    public AdvisoryWrapper addRevisionHistoryEntry(CreateAdvisoryRequest changedCsafJson) {
+
+        return this.addRevisionHistoryEntry(changedCsafJson.getSummary(), changedCsafJson.getLegacyVersion());
+    }
+    public AdvisoryWrapper addRevisionHistoryEntry(String summary, String legacyVersion) {
+
+
+        ArrayNode historyNode = getOrCreateHistoryNode();
+
+        if (getVersioningStrategy().getVersioningType() == VersioningType.Semantic &&  isPrerelease()) {
+            ObjectNode entry = historyNode.addObject();
+            entry.put("summary", summary);
+            entry.put("legacy_revision", legacyVersion);
+            entry.put("number", this.getDocumentTrackingVersion());
+            entry.put("date", this.getDocumentTrackingCurrentReleaseDate());
+        } else {
+            ObjectNode latestEntry = getLatestEntryInHistoryAfterPrerelease(historyNode);
+            if (latestEntry == null) {
+                latestEntry = historyNode.addObject();
+            }
+            latestEntry.put("summary", summary);
+            latestEntry.put("legacy_revision", legacyVersion);
+            latestEntry.put("number", this.getDocumentTrackingVersion());
+            latestEntry.put("date", this.getDocumentTrackingCurrentReleaseDate());
+        }
+        return this;
+    }
+
+    public void removeAllRevisionHistoryEntries() {
+
+        ArrayNode historyNode = getOrCreateHistoryNode();
+        historyNode.removeAll();
+    }
+
+
+    public void removeAllPrereleaseVersions() {
+
+        if (getVersioningStrategy().getVersioningType() == VersioningType.Semantic &&  isPrerelease()) {
+            ArrayNode historyNode = getOrCreateHistoryNode();
+            historyNode.removeAll();
+        }
+    }
+
+
+    private ObjectNode getLatestEntryInHistoryAfterPrerelease(ArrayNode historyNode) {
+
+        if (historyNode.size() > 0) {
+
+            ObjectNode node = (ObjectNode) historyNode.get(historyNode.size() - 1);
+            if (getLastVersion().equals(node.get("number").asText())) {
+                return null;
+            } else {
+               return node;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    public boolean isPrerelease() {
+
+        Semver version = new Semver(this.getDocumentTrackingVersion());
+        return (version.getMajor() < 1)
+                || ((version.getMajor() == 1) && (version.getMinor() == 0) && (version.getPatch() == 0));
```

Comment:
Are we really in pre-release if the version is 1.0.0 ?

Reply:
Yes, we are in the prerelease workflow.
CreateNewCsaf starts with 1.0.1

Reply:
When I read "isPrerelease" I think of "are we not in state published yet", which could also be '2.1.1'. Maybe we should rename the method to something like "isInitialVersion" or "isVersionInitialPublicationOrEarlier"?

Reply:
a rename it to  beforeInitialPublicRelease like defined in the csaf specification